src/stavenote.ts

```diff
@@ -1207,6 +1211,12 @@ export class StaveNote extends StemmableNote {
       this.setStem(new Stem(stemOptions));
     }
 
+    // If we will render a flag, we shorten the stem so that the tip
+    // does not poke through the flag.
+    if (this.shouldDrawFlag() && this.stem) {
```

Comment:
It seems like `shouldDrawFlag` should also check for `this.stem` -- then you can eliminate the `this.stem? || 0` conditionals above too, no?

Reply:
I made these changes but am running into this issue:

TS compiler complains that this.stem can still be undefined (even though I checked in shouldDrawFlag()). So I add a non-null assertion (!). 

```
    if (this.shouldDrawFlag()) {
      const { y_top, y_bottom } = this.getNoteHeadBounds();
      const noteStemHeight = this.stem!.getHeight();

```

But then our eslint rules warn that:

```
vexflow/src/stavenote.ts
  1172:30  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion
```

I see that `// eslint-disable-next-line` is used in some places. Is this a good place to disable eslint? We know `this.stem` is defined at this point, due to `shouldDrawFlag()`.

Anyways, I'll push my changes for your review.

Reply:
Okay, let's leave it like this.